lib/dns.js

```diff
@@ -185,16 +192,19 @@ function onlookupservice(err, host, service) {
 
 // lookupService(address, port, callback)
 exports.lookupService = function(host, port, callback) {
-  if (arguments.length !== 3)
-    throw new Error('Invalid arguments');
+  if (arguments.length !== 3) {
+    throw new TypeError(`Expected 3 arguments, got ${arguments.length}`);
+  }
 
-  if (cares.isIP(host) === 0)
+  if (isIP(host) === 0) {
     throw new TypeError('"host" argument needs to be a valid IP address');
+  }
 
-  if (port == null || !isLegalPort(port))
+  if (port == null || !isLegalPort(port)) {
     throw new TypeError(`"port" should be >= 0 and < 65536, got "${port}"`);
+  }
 
-  port = +port;
+  port = Number(port);
```

Comment:
Well, unary `+` and `Number` do the same thing but `Number` is more explicit and obvious. Not everyone is familiar with things like this so when there is no doubt about performance issues I'd rather refactor to cleaner code.

I'm also not sure about the `>>>`s to convert to integer numbers. We use them for the options object in this file and I wonder if I shouldn't convert them to `Math.floor` since it's more obvious and it's not a performance sensitive area.